core/table.ts

```diff
@@ -1,63 +1,165 @@
+
+/**
+ * @hidden
+ */
+export const DistStyleType = ["even", "key", "all"] as const;
+/**
+ * Valid types for setting the distribution style for Redshift tables.
+ *
+ * View the [Redshift documentation](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_examples.html#r_CREATE_TABLE_NEW-diststyle-distkey-and-sortkey-options) for more information.
+ */
+export type DistStyleType = typeof DistStyleType[number];
+
+/**
+ * @hidden
+ */
+export const SortStyleType = ["compound", "interleaved"] as const;
+/**
+ * Valid types for setting the sort style for Redshift tables.
+ *
+ * View the [Redshift documentation](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_examples.html#r_CREATE_TABLE_NEW-diststyle-distkey-and-sortkey-options) for more information.
+ */
+export type SortStyleType = typeof SortStyleType[number];
+
+/**
+ * Redshift specific warehouse options.
+ */
+export interface IRedshiftOptions {
+  /**
+   * Sets the DISTKEY property when creating tables.
+   *
+   * For more information, read the AWS documentation [here](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_examples.html#r_CREATE_TABLE_NEW-diststyle-distkey-and-sortkey-options).
+   */
+  distKey?: string;
+  distStyle?: string;
+  sortKeys?: string[];
+  sortStyle?: string;
+  bind?: boolean;
 }
-export enum DistStyleTypes {
-  EVEN = "even",
-  KEY = "key",
-  ALL = "all"
+
+/**
+ * Options for creating tables within Azure SQL Datawarehouse projects.
+ */
+export interface ISQLDataWarehouseOptions {
+  /**
+   * The distribution option value.
+   *
+   * For more information, read the Azure documentation [here](https://docs.microsoft.com/en-gb/sql/t-sql/statements/create-table-as-select-azure-sql-data-warehouse?view=aps-pdw-2016-au7#examples-for-table-distribution).
+   */
+  distribution?: string;
 }
-export enum SortStyleTypes {
-  COMPOUND = "compound",
-  INTERLEAVED = "interleaved"
+
+/**
+ * Options for creating tables within BigQuery projects.
+ */
+export interface IBigQueryOptions {
+  /**
+   * The key for partitioning the table by, typically a timestamp or date.
+   *
+   * For more information, read the BigQuery documentation [here](https://cloud.google.com/bigquery/docs/partitioned-tables#next_steps).
+   */
+  partitionBy?: string;
 }
 
-export const ignoredProps: {
-  [tableType: string]: Array<keyof dataform.ITable>;
-} = {
-  [TableTypes.INLINE]: [
-    "bigquery",
-    "redshift",
-    "sqlDataWarehouse",
-    "preOps",
-    "postOps",
-    "actionDescriptor",
-    "disabled",
-    "where"
-  ]
-};
-
-type ValueOf<T> = T[keyof T];
-export type TContextable<T> = T | ((ctx: TableContext) => T);
-export type TableType = ValueOf<TableTypes>;
-
-export interface TConfig {
+/**
+ * General options that can be provided to a table.
+ */
+export interface ITableConfig extends ICommonOutputConfig {
+  /**
+   * The type of the table. For more information, check out the documentation [guides](guides).
+   */
   type?: TableType;
```

Comment:
I think get rid of the second sentence here. `TableType` already links to the relevant section.